### PR TITLE
Fixed Reload bug with Browser Session Storage

### LINKED ISSUE #12: Reload does not retain current Competition (Admin Dashboard)
If you are on a page (say check in) and then you refresh/reload the page. The selected competition disappears (and consequently so does the side navbar) but the current page still remains.

To recreate:

1. Go to admin dashboard
2. Select a competition
3. Go to scanner -> check in
4. refresh page
5. :( Problem

### PR DESCRIPTION
**Pull request by AlexanderWangY**
Closes #12 

Now, on competition selection, it saves your entry into the browser session so when you reload, you pop back onto that page!

### CODE REVIEW

File: components/admin/AdminShell.tsx [hunk 1]
```diff
@@ -55,12 +60,20 @@ export default function AdminShell({
...
   const { data } = useSWR<Competition[]>('/api/comp', fetcher, {
     fallbackData: [],
   });
 
-  const [comp, setComp] = useState<string | null>(null);
+  useEffect(() => {
+    const selectedComp = sessionStorage.getItem('selectedComp');
+
+    // Check if the selected competition still exists
+    if (selectedComp && data?.find((c) => c.code === selectedComp)) {
```
Comment (RobertConde) [inline components/admin/AdminShell.tsx:73]: We don't care about the value here. We only need to know if there exists some item that matches the predicate. Use `.some(...)`.

File: components/admin/AdminShell.tsx [hunk 2]
```diff
@@ -165,7 +187,11 @@ export default function AdminShell({
                 active={pathname === '/admin/comp/events'}
               />
 
-              <NavLink label='Scanner' leftSection={<IconQrcode size='1rem' />}>
+              <NavLink
+                label='Scanner'
+                leftSection={<IconQrcode size='1rem' />}
+                defaultOpened={pathname.startsWith('/admin/comp/scan')}
```
Comment (RobertConde) [inline components/admin/AdminShell.tsx:193]: 🫶

File: components/admin/AdminShell.tsx [hunk 3]
```diff
@@ -103,7 +116,15 @@ export default function AdminShell({
             placeholder='Select a competition'
             data={data?.map((c) => ({ value: c.code, label: c.name }))}
             value={comp}
-            onChange={(comp) => setComp(comp)}
+            onChange={(comp) => {
+              setComp(comp);
+
+              if (comp) {
+                sessionStorage.setItem('selectedComp', comp);
```
Comment (RobertConde) [inline components/admin/AdminShell.tsx:123]: nit: I had braces (I'm going to add an eslint / prettier rule)

File: components/admin/AdminShell.tsx [hunk 4]
```diff
@@ -103,7 +116,15 @@ export default function AdminShell({
             placeholder='Select a competition'
             data={data?.map((c) => ({ value: c.code, label: c.name }))}
             value={comp}
-            onChange={(comp) => setComp(comp)}
+            onChange={(comp) => {
```
Comment (RobertConde) [inline components/admin/AdminShell.tsx:119]: My fault but avoid confounding variable names. Use `c` here (notice, `comp` is also a variable).

**Review by RobertConde — CHANGES_REQUESTED** [review]
slay

File: components/admin/AdminShell.tsx (hunk 1 above)
Comment (AlexanderWangY) [reply components/admin/AdminShell.tsx]: sounds good :+1:

**Review by RobertConde — APPROVED** [review]

**Pull request merged**